### PR TITLE
Using the new shell API and moving nodes out of this repo

### LINKED ISSUE #17: Migrate Shells object to use new Shell low-level API.
A new Shell low-level API was merged in the base topology framework in https://github.com/HPENetworking/topology/commit/06f9d9874dbc6328f8e406737c8a85cdf4ed2da0

This platform engine should migrate it's shell objects to use this new API.

### LINKED ISSUE #18: Place node-specific code in separate repos
The implementation of nodes for this platform engine resides in this very same repo, they should be moved to completely separate the node development from this.

### PR DESCRIPTION
**Pull request by ocelotl**
Please copy the following repos into HPENetworking:

https://github.com/ocelotl/topology_docker_p4switch
https://github.com/ocelotl/topology_docker_openvswitch
https://github.com/ocelotl/topology_docker_ryu
https://github.com/ocelotl/topology_docker_openswitch

Once that is done, please change the requirements.dev.txt file to match the new repos.

This closes #18. This closes #17.

### CODE REVIEW

**Comment by dajose** [comment]
Repos moved to:

https://github.com/HPENetworking/topology_docker_p4switch
https://github.com/HPENetworking/topology_docker_openvswitch
https://github.com/HPENetworking/topology_docker_ryu
https://github.com/HPENetworking/topology_docker_openswitch

I'll complete the rest tomorrow

File: requirements.dev.txt [hunk 1]
```diff
@@ -6,3 +6,9 @@ sphinx
 sphinx_rtd_theme
 sphinxcontrib-plantuml
 autoapi
+ipdb
+-e git+https://github.com/HPENetworking/topology.git@master#egg=topology
```
Comment (dajose) [inline requirements.dev.txt]: shouldn't this be on requirements.txt ?

File: requirements.dev.txt [hunk 2]
```diff
@@ -6,3 +6,9 @@ sphinx
 sphinx_rtd_theme
 sphinxcontrib-plantuml
 autoapi
+ipdb
```
Comment (dajose) [inline requirements.dev.txt]: hehe you do love this module

Comment (ocelotl) [inline requirements.dev.txt]: Damn right.

File: requirements.dev.txt (hunk 1 above)
Comment (ocelotl) [inline requirements.dev.txt]: Should be, but the shell API is still not included in the topology release.

File: requirements.dev.txt [hunk 3]
```diff
@@ -6,3 +6,9 @@ sphinx
...
+-e git+https://github.com/HPENetworking/topology.git@master#egg=topology
+-e git+https://github.com/ocelotl/topology_docker_openswitch.git@master#egg=topology_docker_openswitch
+-e git+https://github.com/ocelotl/topology_docker_host.git@master#egg=topology_docker_host
```
Comment (dajose) [inline requirements.dev.txt]: shouldn't we leave docker_host as a default node? (leave it on this repo? ) @carlos-jenkins

Comment (ocelotl) [inline requirements.dev.txt]: It could be, but it kinda defeats the purpose of removing all the node code out of this repo...

**Comment by ocelotl** [comment]
Updated branch to include host node again.

File: lib/topology_docker/nodes/host.py [hunk 4]
```diff
@@ -16,32 +16,8 @@
 # under the License.
 
 """
-Simple host Topology Docker Node using Ubuntu.
+topology_docker.nodes module entry point.
 """
 
 from __future__ import unicode_literals, absolute_import
 from __future__ import print_function, division
-
```
Comment (dajose) [inline lib/topology_docker/nodes/host.py]: wait... this shouldn't be removed

**Pull request closed without merging**